src/lib/comms/include/sol-netctl.h

```diff
@@ -153,6 +153,57 @@ enum sol_netctl_service_state {
     SOL_NETCTL_SERVICE_STATE_REMOVE,
 };
 
+/**
+ * @brief method of proxy generated for a network link.
+ *
+ */
+enum sol_proxy_method {
+    SOL_PROXY_METHOD_DIRECT,
+    SOL_PROXY_METHOD_AUTO,
+    SOL_PROXY_METHOD_MANUAL,
+};
+
+/**
+ * @brief Structure to represent a network proxy.
+ *
+ * This struct contains the necessary information to present a
+ * network proxy for user.
+ */
+struct sol_proxy {
+    enum sol_proxy_method method;
```

Comment:
Of course we can change . What kind of change do you refer to ?
BTW, which API version define here?